src/blocks/scratch3_data.js

```diff
@@ -122,6 +123,34 @@ class Scratch3DataBlocks {
         return list.value[index - 1];
     }
 
+    getItemNumOfList (args, util) {
+        const item = args.ITEM;
+        const list = util.target.lookupOrCreateList(
+            args.LIST.id, args.LIST.name);
+
+        // Go through the list items one-by-one using Cast.compare. This is for
+        // cases like checking if 123 is contained in a list [4, 7, '123'] --
+        // Scratch considers 123 and '123' to be equal.
+        for (let i = 0; i < list.value.length; i++) {
+            if (Cast.compare(list.value[i], item) === 0) {
+                return i + 1;
+            }
+        }
+
+        // We don't bother using .indexOf() at all, because it would end up with
```

Comment:
It wouldn't be any "faster" performance wise, the for loop will win here,  but you could probably use `findIndex` to simplify the code here:

```js
return 1 + list.value.findIndex( cmp => Cast.compare(cmp, item) === 0 );
```

Reply:
Oh, that's definitely clever. A related change would be below, for `listContainsItem`:

```js
return list.value.some(cmp => Cast.compare(cmp, item) === 0);
```

I'll probably leave this as is for the (probably very minor) performance boost (and so I don't modify the other function's code in this PR), but I can make that change if wanted. Array functions like `findIndex` and `some` are very nice!